### ISSUE TITLE
Very low test accuracy with C dataset

### ISSUE THREAD

**Issue by dhas** [tech-srl/code2vec#84]
Hi @urialon 

I've been trying to train code embeddings for a dataset of C sources using your code2vec technique. I collected a (quite small) dataset of 2 sets of Linux drivers. I then parsed it using [astminer](https://github.com/JetBrains-Research/astminer) and used your preprocess.sh to create the necessary c2v files. The result is a training set of ~6k functions and a validation set of ~1.5k functions.  I then used it to train a keras code2vec model with all hyperparameters intact except for the batch size, which I reduced to 64.  Upon training for 30 epochs, I find that the top1 validation accuracy is a very low 0.08 (top10 accuracy 0.14). 

I must admit that I've been only 'hacking' your code and haven't dug in deeply yet, but such bad results makes me wonder whether my improvised parsing pipeline is correct. I'm attaching
[accuracy_issue.zip](https://github.com/tech-srl/code2vec/files/4747409/accuracy_issue.zip) with the following files:
1. The directory data/linux-drivers-small, which contains all .c2v files required for training
2. training_log.txt which contains stdout logs from training

I hope you can give me a quick way ahead while I continue hunting deeper.

Thanks

**Comment by urialon** [comment]
Hi @dhas,
Thank you for your interest in code2vec.

I haven't looked at the attached file, but it is very likely that 6k
examples are too few.
You can try our code2*seq* model which is much more data-efficient.
Additionally, please use our default TensorFlow model, as not all features
are implemented in the Keras version.

Best,
Uri

**Comment by dhas** [comment]
Hi @urialon 

Thanks for your quick response. I was expecting this answer from you. I'd also tried with another dataset with ~60k training samples and ~14k validation samples. Even here, the validation accuracy was similarly low. Is this also a size that you would consider too low? If not, is a better set of hyperparameters the solution?

**Comment by urialon** [comment]
Hi,
It's difficult to tell. In the paper, we trained the model on 14M examples.
I have no idea "how few is too few".

You can try our code2seq model which is much more data-efficient and see if
the results improve.

**Comment by dhas** [comment]
Ok thanks
